src/api/spec/controllers/webui/comments_controller_spec.rb

```diff
@@ -55,6 +54,47 @@
       end
     end
 
+    context 'with a commentable that does not exist' do
+      context 'of a project' do
+        let(:commentable) { project }
+
+        before do
+          params = { comment: { body: 'This project is AWESOME!' },
+                     commentable_type: commentable.class, commentable_id: -commentable.id }
+          post :create, params: params
```

Comment:
We are testing the action `POST #create` in all these tests. I would have preferred to replace all the lines:
```
post :create, params: params
```
with
```
subject { post :create, params: comment_params }
```
, as it was used previously.

Later I would have moved out all the `subject {...}` calls to only one line, after the `describe 'POST #create' do` block definition. And then adapt the tests, to use `subject`.

The reasoning is that in the `describe 'POST #create' do` block all the tests share the same subject. They always make a call to `post :create`.

This could be done later. I will not stop this PR from being merged only for this reason. What do you think?

Reply:
I added a comment with a proposal of spec refactoring. Let me know what do you think

Reply:
It seems that yes, the first step of the three has been addressed. I modified my previous comment to point out what are the three steps I'm talking about.

Reply:
Well, I could not do it **exactly** as you propose, because the spec uses two different subjects: the immediate one `subject!` and a delayed one `subject`.

That last one is needed for the use case where we're expecting an error raise (see [here](https://github.com/openSUSE/open-build-service/pull/11527/files#diff-bee2469df30c00bd819919cc1cedb771cd5d830c614b8e1e456adb7717475bacR113), I need the subject to trigger inside the `expect { ... }` block to catch the error, otherwise, the spec fails).

So I cannot move them all to just one `subject` block...

Do you agree?

Reply:
True. The test of checking for errors is the only one that requires `subject` (without `!`). I'm fine with the given solution, then.